### ISSUE TITLE
[no-empty-function] False-positive on private constructor

### ISSUE THREAD

**Issue by Dionysusnu** [typescript-eslint/typescript-eslint#4778]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

[playground](https://typescript-eslint.io/play/#ts=4.6.2&sourceType=module&code=MYGwhgzhAEBmD29oG8CwAoaXoAcBOAlgG5gAuAptMPAHYSl4Cuwp8eAFAJQoC+GPQA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AQPY4C2mUWAZgK4FIRpkEgBcIcKKZKEAF8gA&tsConfig=N4XyA)

```TS
class foo {
    private constructor() {}
}
```

**Expected Result**
This should not be an error. Constructors must have the `{}`, and removing the constructor is an unwanted change in behaviour, because other code would then be able to construct the class.

**Actual Result**
The rule reports a warning on the constructor, specifically on the `{}`.

**Additional Info**
N/A

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.17.0` |
| `@typescript-eslint/parser`        | `5.17.0` |
| `TypeScript`                       | `4.6.2` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `16.13.1`, also happens in browser using playground above |

**Comment by Dionysusnu** [comment]
Sorry, forgot to check rule options.
Add it as
```json
{
  "rules": {
    "@typescript-eslint/no-empty-function": ["error", { "allow": "private-constructors" }]
  }
}
```
